extensions/wikia/ThemeDesigner/ThemeSettingsPersistence.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+class ThemeSettingsPersistence {
+	/** @var ThemeSettings $themeSettings */
+	private $themeSettings;
+	/** @var int $cityId */
+	private $cityId;
+
+	public function __construct( ThemeSettings $themeSettings ) {
+		$this->themeSettings = $themeSettings;
+		$this->cityId = $themeSettings->getCityId();
+	}
+
+	public function saveSettings( $settings ) {
+		global $wgUser;
+		
+		$oldBackgroundFile = $this->saveImage(
+			$settings,
+			'background-image',
+			ThemeSettings::BackgroundImageName,
+			[
+				'width' => 120,
+				'height' => 65
+			],
+			true
+		);
+
+		$oldCommunityHeaderFile = $this->saveImage(
+			$settings,
+			'community-header-background-image',
+			ThemeSettings::CommunityHeaderBackgroundImageName,
+			[
+				'width' => 120,
+				'height' => 33
+			]
+		);
+
+		$oldWordmarkFile = $this->saveImage(
+			$settings,
+			'wordmark-image',
+			ThemeSettings::WordmarkImageName
+		);
+
+		$oldFaviconFile = $this->saveImage(
+			$settings,
+			'favicon-image',
+			ThemeSettings::FaviconImageName,
+			[],
+			false,
+			function () {
+				Wikia::invalidateFavicon();
+			}
+		);
+
+		$reason = wfMessage( 'themedesigner-reason', $wgUser->getName() )->escaped();
+
+		// update history
+		$wikiFactoryHistory = WikiFactory::getVarValueByName(
+			ThemeSettings::WikiFactoryHistory,
+			$this->cityId
+		);
+
+		if ( !empty( $wikiFactoryHistory ) ) {
+			$history = $wikiFactoryHistory;
+			$lastItem = end( $history );
+			$revisionId = intval( $lastItem['revision'] ) + 1;
+		} else {
+			$history = [];
+			$revisionId = 1;
+		}
+
+		// SUS-2942: get not explicitly set values from old settings
+		$oldSettings = $this->themeSettings->getSettings();
+		$settings = array_merge( $oldSettings, $settings );
+		// update WF variable with current theme settings
+		WikiFactory::setVarByName( ThemeSettings::WikiFactorySettings, $this->cityId, $settings, $reason );
+
+		// add entry
+		$history[] = [
+			'settings' => $settings,
+			'author' => $wgUser->getName(),
+			'timestamp' => wfTimestampNow(),
+			'revision' => $revisionId,
+		];
+
+		// limit history size to last 10 changes
+		$history = array_slice( $history, -ThemeSettings::HistoryItemsLimit );
+
+		if ( count( $history ) > 1 ) {
+			for ( $i = 0; $i < count( $history ) - 1; $i++ ) {
+				if ( isset( $oldFaviconFile ) && isset( $history[$i]['settings']['favicon-image-name'] ) ) {
+					if ( $history[$i]['settings']['favicon-image-name'] == ThemeSettings::FaviconImageName ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)